scripts/import_dvv.sql

```diff
@@ -21,21 +21,134 @@ insert into jkr_osoite.katu (katunimi_fi, katunimi_sv, kunta_koodi)
 select distinct
     "kadunnimi suomeksi" as katunimi_fi,
     "kadunnimi ruotsiksi" as katunimi_sv,
-    sijainti_kunta as kunta_koodi
+    sijainti_kunta as kunta_koodi -- names may be null. sijaintikunta is never null.
 from jkr_dvv.osoite
-where "kadunnimi suomeksi" is not null or "kadunnimi ruotsiksi" is not null  -- not all places have street address
-on conflict do nothing;
+on conflict do nothing; -- create one empty street for each kunta
 
 -- Insert addresses to jkr.osoite
--- Assume Finnish kadunnimi is given. Some (two) entries might have errors because of this.
 insert into jkr.osoite (osoitenumero, katu_id, rakennus_id, posti_numero)
 select
     katu_numero as osoitenumero,
-    (select id from jkr_osoite.katu where osoite."kadunnimi suomeksi" = katu.katunimi_fi and osoite.sijainti_kunta = katu.kunta_koodi) as katu_id,
+    case when (osoite."kadunnimi suomeksi" is not null)
+        then (select id from jkr_osoite.katu where osoite."kadunnimi suomeksi" = katu.katunimi_fi and osoite.sijainti_kunta = katu.kunta_koodi)
```

Comment:
> Doesn't this return multiple rows in these cases?

Good point. For most rows, we only have Finnish name, so there seem to be no cases where we have different translations while the other name is identical *and* not null. But your suggestion should handle these cases too.

Reply:
Hm, the suggestion actually revealed a problem in the original katu import.

In DVV data, we have rows such as
```
 katunimi_fi | katunimi_sv  | kunta_koodi 
-------------+--------------+-------------
 Niemeläntie | Niemelävägen | 504
 Niemeläntie |              | 504
```

which conflicts with the `katunimi_fi_idx` unique constraint.

So we have to decide if we allow multiple streets with same Finnish or Swedish names in the same kunta, if the combinations are different, or we skip some duplicate lines such as this in DVV data.

This is why I did preferentially
```
(select id from jkr_osoite.katu where osoite."kadunnimi suomeksi" = katu.katunimi_fi and osoite.sijainti_kunta = katu.kunta_koodi)
```

and match with the Swedish name *only* if the Finnish name is null. This way, such incomplete rows will be mapped to the correct more complete rows in DVV data.

Reply:
Yep, will have to retain my original case formulation, because we have to match rows with only Finnish *or* Swedish names to addresses that may have *both* Finnish *and* Swedish names, since we cannot have duplicates. So we have to do the matching based on one field at a time.

Reply:
So, your original comment is taken into account by the unique constraints `katunimi_fi_idx` and `katunimi_sv_idx`. Fixed the street import in https://github.com/GispoCoding/jkr-core/pull/20/commits/11e77a606a0624b460ef80173846b1b9324beb4c

Reply:
This is good enough.